[kyueran] iP

kyueran opened the pull request:
# Duke-Gerty

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke-Gerty frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use
- All you need to do is,

download it from [here](https://github.com/kyueran/ip).
1. double-click it.
2. add your tasks.
3. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

#

You should edit the file path in the main method to load your tasks from a custom file! Here's the main method:
```ruby
public static void main(String[] args) {
    new Duke("data/tasks.txt").run();
}
```

Commit:
Add Gradle support

kyueran committed:
Added Level 1 features, Greet, Echo, Exit

kyueran committed:
Added Level 2 features, Add, List

kyueran committed:
Added Level 3 features, Mark as Done

kyueran committed:
Added Level 4 features, ToDo, Event, Deadline

kyueran committed:
Added Automated Text UI Testing

kyueran committed:
Added Level 5 features, Error handling

kyueran committed:
Added Level 6 feature, Delete, updated input.txt for testing

kyueran committed:
Added A-Enums feature, use Enums

kyueran committed:
Added Level 7 feature, Save

kyueran committed:
Added Level 8 feature, Dates and Times

kyueran committed:
Merge branch 'branch-Level-7'

kyueran committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Task.java

kyueran committed:
Added A-MoreOOP, Use More OOP, Minimal and Stretch goals fulfilled

kyueran committed:
Added A-Packages feature, organize into packages and fix EXPECTED.TXT text testing

kyueran committed:
Merge commit '556af3f47a96b32898ab4cdbd65b16486a4871e8'

kyueran committed:
Added A-Gradle feature, use gradle

kyueran committed:
Added A-JUnit feature, add JUnit Tests

kyueran committed:
Add A-Jar feature, create a JAR file

kyueran committed:
Add A-JavaDoc feature

kyueran committed:
Add A-CodingStandard feature, follow the coding standard

kyueran committed:
Add Level-9 feature, Find

kyueran committed:
Merge branch 'branch-A-CodingStandard'

kyueran committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/parser/Parser.java
#	src/main/java/storage/Storage.java
#	src/main/java/tasks/TaskList.java
#	src/main/java/ui/Ui.java

kyueran committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/tasks/TaskList.java

kyueran committed:
Add A-Checkstyle feature, pass all checkstyle tests

kyueran committed:
Added Level-10 feature, GUI

unfazing commented on src/main/java/duke/commands/ByeCommand.java:
```diff
@@ -0,0 +1,19 @@
+    @Override
+    public void execute(TaskList tl, Ui ui, Storage s) {
+        ui.display("Bye. Hope to see you again soon!");
+    }
```
You can consider inserting a newline between functions 😄

unfazing commented on src/main/java/duke/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,35 @@
+    public DeadlineCommand(String desc, String by) {
+        this.desc = desc;
+        this.by = by;
+    }
```
Likewise, you can consider inserting a newline between the functions for readability and consistency across all your code (All your command classes.)

kyueran replied:
Ok will take note!

unfazing commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,111 @@
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] arr = fullCommand.split(" ", 2);
+        String commandType = arr[0].toLowerCase();
+        if (commandType.equals(CommandType.BYE.toString())) {
```
You can consider using a `switch` block for this section 😄

kyueran replied:
I tried using switch but I encountered some issues using switch with enum because the switch cases need to be constants but my enum toString method somehow does not evaluate to a constant at compile time.

unfazing commented on src/main/java/duke/enums/CommandType.java:
```diff
@@ -0,0 +1,39 @@
+        this.e = e;
+    }
+
+    public DukeException getErr() {
```
You can consider adding JavaDocs for this public method.

unfazing commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,197 @@
+        }
+    }
+
+    /**
```
You can consider including the @throws section  in the JavaDocs to include situations when the method throws an Exception

kyueran replied:
Yup thanks! I actually thought the style guide meant only header comment is necessary and not the entire javadocs

unfazing commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,197 @@
+        }
+    }
+
+    /**
```
JavaDocs does not contain @param section.

unfazing commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,197 @@
+        appendToFile(taskText);
+    }
+
+    /**
```
Javadocs does not contain @param section

unfazing commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,197 @@
+        modifyLine(oldText, newText);
+    }
+
+    /**
```
JavaDocs does not contain @param section

unfazing commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,93 @@
+        this.description = description;
+        this.isDone = false;
+    }
+    public String getStatusIcon() {
```
Missing JavaDocs for the public methods getStatusIcon, markAsDone and unmarkAsDone

unfazing commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,90 @@
+ */
+public class TaskList {
+    private List<Task> tasks;
+    public TaskList() {
```
JavaDocs missing for constructors and addTask, removeTask methods.

unfazing commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,90 @@
+        tasks.remove(i);
+    }
+
+    /**
```
JavaDocs missing @param (and @return) section for the below methods.

unfazing commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,80 @@
+
+    private String response = showWelcome();
+
+    /**
```
JavaDoc does not contain @param section

unfazing commented on src/main/java/gui/DialogBox.java:
```diff
@@ -0,0 +1,72 @@
+        setAlignment(Pos.TOP_LEFT);
+    }
+
+    public static DialogBox getUserDialog(String text, Image img) {
```
Missing JavaDocs for the getUserDialog and getDukeDialog methods.

unfazing commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,86 @@
+import duke.ui.Ui;
+
+/**
+ * duke.Duke class that runs main chatbot logic.
```
You can consider adding an @author tag for all your files.

kyueran replied:
Ok thanks!

unfazing reviewed (commented):
The code is easily understandable and readable. Apart from some minor styling and JavaDoc issues, it LGTM. 😄

kyueran committed:
Add assert statements to code

Currently there are no assert statements to verify assumptions about the progam at runtime

Assertions can help detect bugs and invalid states in the code to assist in debugging

Let's add assertions to document important assumptions that should hold at various points in the code.

kyueran committed:
Merge pull request #1 from kyueran/A-Assertions

Add assert statements to code

kyueran committed:
Improve code quality

kyueran committed:
Merge pull request #2 from kyueran/A-CodeQuality

Improve Code Quality

kyueran committed:
Merge branch 'master' of https://github.com/kyueran/ip

kyueran committed:
Add functionality to deal with duplicates

Currently new ToDo, Deadline and Event tasks are added to the task list and storage file even if they are duplicates.

We should avoid doing this to clutter the user's tasklist with duplicate tasks that the user had forgotten they added previously

Let's prevent the addition of duplicate tasks and display to the user in the GUI that the task was not added as it already exists in the tasklist

We still allow user to add duplicate of the same task if one is marked and one is unmarked as the user might want to keep track of the task in both states

kyueran committed:
Merge pull request #3 from kyueran/C-DetectDuplicates

Add functionality to deal with duplicates

kyueran committed:
Merge branch 'master' of https://github.com/kyueran/ip

kyueran committed:
Add A-UserGuide

kyueran committed:
Update README.md

kyueran committed:
Update README.md

kyueran committed:
Update README.md
